rest-service/manager_rest/rest/resources_v3_1/manager.py

```diff
@@ -223,12 +232,68 @@ def _get_ca_cert_id(self, sm, _manager):
             return existing_managers_certs.pop()
 
 
-class RabbitMQBrokers(SecuredResource):
+class RabbitMQBrokers(ManagementResource):
     @exceptions_handled
     @marshal_with(models.RabbitMQBroker)
     @paginate
     @authorize('broker_get')
     def get(self, pagination=None):
+        """List brokers from the database."""
         return get_storage_manager().list(
             models.RabbitMQBroker,
         )
+
+    @exceptions_handled
+    @authorize('broker_add')
+    @marshal_with(models.RabbitMQBroker)
+    def post(self):
+        """Add a broker to the database."""
+        _broker = rest_utils.get_json_and_verify_params({
+            'name': {'type': unicode},
+            'address': {'type': unicode},
+            'port': {'type': int, 'optional': True},
+            'networks': {'type': dict, 'optional': True},
+        })
+        sm = get_storage_manager()
+
+        # Get the first broker in the list to get the ca_cert and credentials
+        first_broker = sm.list(models.RabbitMQBroker, None, None).items[0]
+
+        if not _broker.get('networks'):
+            _broker['networks'] = {'default': _broker['address']}
+
+        new_broker = models.RabbitMQBroker(
+            name=_broker['name'],
+            host=_broker['address'],
+            management_host=_broker['address'],
+            port=_broker.get('port'),
+            networks=_broker['networks'],
+            username=first_broker.username,
+            password=first_broker.password,
+            ca_cert=first_broker.ca_cert,
+        )
+        result = sm.put(new_broker)
+        current_app.logger.info('Broker added successfully')
+        update_agents(sm)
+        return result
+
+    @exceptions_handled
+    @authorize('broker_delete')
+    @marshal_with(models.RabbitMQBroker)
+    def delete(self):
```

Comment:
unfortunate that this is using the same weird handling as `Managers`, ie. request body contains the id to be deleted, as opposed to `DELETE /brokers/broker-123`. You think perhaps we can make this that, or do you want consistency with managers (but not with anything else in the system)?
Either way, we should also change managers to be the other way

Reply:
Agreed, I was thinking we should change it to be properly restful.

Reply:
This will go into a separate PR